### DISCUSSION TITLE
Backup path - File index by date stamp

### DISCUSSION THREAD

**Discussion by penguinpages** [webmin/webmin#1871] [Ideas]
### 

Have a lab that is using webmin for DNS and DHCP.. Trying to help with this but one thing that has me confused is the backup within webmin of configuration files.    Ex: make backup of DNS / DHCP before changes are made.

But backup is single simple path with no option to embeed date / time stamp into file.  As such .. every backup overwrites the previous one and so role back is ... not really helpful

Can we get module enhancement to either allow shell name of path location that can take date string at end:  

Ex: /home/dnsuser/webmin_dns_dhcp_backup_'date +%y%m%d%H%M%S`'

or set as option within the GUI to apend date or some kind of index number.

**Comment by iliaross** [comment]
Hello,

If you're talking about `Webmin ⇾ Backup Configuration Files` module then you can enable **strftime substitution** in module's config (cog button on top-left of the module index page).

**Comment by penguinpages** [comment]
I tried that..  Assuming it was what was needed:

 webmin_dns_dhcp_backup -> Local File "/home/dnsuser/webmin_dns_dhcp_backup" 

<gear> -> Do strftime substitution of backup destinations? -> yes

Return to backup:
Run backup

###
Starting backup of module configuration files to /home/dnsuser/webmin_dns_dhcp_backup ..
.. complete. Final backup size was 4.79 KiB, containing 26 files.
###

```
dnsuser@ps-dns-01:~$ ls
nohup.out   webmin_dns_dhcp_backup
dnsuser@ps-dns-01:~$

```

**Comment by iliaross** [comment]
I have just tried this:

<img width="1208" alt="image" src="https://user-images.githubusercontent.com/4426533/227545618-8a283839-0139-478d-8026-419268d9de01.png">

.. and got this:

<img width="1209" alt="image" src="https://user-images.githubusercontent.com/4426533/227545850-cea86595-0d6e-4add-975d-259be5e1b641.png">

Didn't it work for you?

**Comment by penguinpages** [comment]
That set of variables at end fixed this. 

%y-%d-%H:%M:%S


```
-rw------- 1 root    root    4.8K Mar 24 14:34 webmin_dns_dhcp_backup_23-24-14:34:39
-rw------- 1 root    root    4.8K Mar 24 14:42 webmin_dns_dhcp_backup_23-24-14:42:31
-rw-rw-r-- 1 dnsuser dnsuser  170 Jan 19  2022 .wget-hsts
dnsuser@ps-dns-01:~$
```

IMAO:  checkbox should enable this.. or if its a combination  of that checkbox and adding your desired  variablized file name at end... cool..but link to document or example within UI etc..  Sorry if I missed this documented somewhere within webmin documenation.. but I googled around and did not get a hit.

Thanks

**Comment by iliaross** [comment]
>  or if its a combination of that checkbox and adding your desired variablized file name at end

Yes, it is a combination.

> Sorry if I missed this documented somewhere within webmin documenation

This feature was added after initial Webmin documentation was written. We need to fix that though in the new, upcoming Webmin documentation.

> cool..but link to document or example within UI etc

@jcameron Jamie, let's make **strftime substitution** enabled by default and add a placeholder to the path input with an example. It should resolve most of the common problems with _strftime_ substitutions.

**Comment by jcameron** [comment]
Agreed, we should enable it by default. What do you mean by a placeholder though?

**Reply by iliaross** [reply]
> Agreed, we should enable it by default.

Alright. Both done in [this](https://github.com/webmin/webmin/commit/8694adcf93271c3a3d6b20608913ab85b62e5722) commit.

> What do you mean by a placeholder though?

A placeholder is an attribute specifies a short hint that describes the expected value of an input field:

<img width="910" alt="image" src="https://user-images.githubusercontent.com/4426533/227608266-927eb2ce-fe28-4074-9d5d-fa79632a1b92.png">

**Reply by jcameron** [reply]
Oh nice ... please add that!

**Reply by iliaross** [reply]
Thanks. Already added.
